### PR TITLE
Update Balance design with chart

### PR DESCRIPTION
**Pull request by kosssi**
without balance flag | with balance flag
---------------------|-----------------
capture d ecran 2018-12-04 a 14 23 11 | capture d ecran 2018-12-04 a 14 22 26
capture d ecran 2018-12-04 a 14 24 56 | capture d ecran 2018-12-04 a 14 25 23

### CODE REVIEW

File: src/components/Header/Header.styl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+.HeaderColor_primary
+    background-color var(--primary)
+
+.Header_fixed
+    z-index 1
+    position fixed
+    top 3rem
+    right 0
+    left 221px
```
Comment (ptbrowne) [inline src/components/Header/Header.styl]: Where does this value come from ?

File: src/ducks/balance/Balance.jsx [hunk 2]
```diff
@@ -284,18 +285,29 @@ class Balance extends React.Component {
       settings: settingsCollection,
       transactions: transactionsCollection
     } = this.props
+
+    const withChart = flag('balance-history')
+    const headerColorProp = { color: withChart ? 'primary' : 'default' }
```
Comment (ptbrowne) [inline src/ducks/balance/Balance.jsx]: s/headerColorProp/headerColorProps

File: src/components/Chart/LineChart.jsx [hunk 3]
```diff
@@ -306,15 +306,33 @@ class LineChart extends Component {
   }
 
   render() {
-    const { width, height, gradient, margin } = this.props
+    const { width, height, gradient, margin, container } = this.props
     const { selectedItem, x } = this.state
 
+    const containerWidth = get(container, 'current.offsetWidth')
```
Comment (ptbrowne) [inline src/components/Chart/LineChart.jsx]: Do not use `offsetWidth` inside `render` since this will trigger a reflow : https://gist.github.com/paulirish/5d52fb081b3570c81e3a#box-metrics.

File: src/components/Chart/LineChart.jsx [hunk 4]
```diff
@@ -306,15 +306,33 @@ class LineChart extends Component {
...
+    const isContainerGreater = containerWidth > width
+    const marginAuto = isContainerGreater ? (containerWidth - width) / 2 : 0
+    const positionProps = x < width / 2 ? { left: 0 } : { right: 0 }
+    const tooltipX =
+      x < width / 2
+        ? x + marginAuto + margin.left
+        : isContainerGreater
+          ? -(width - x + marginAuto - margin.right)
+          : -(width - x - (width - containerWidth) - margin.right)
```
Comment (ptbrowne) [inline src/components/Chart/LineChart.jsx]: I find this part hard to follow, can this be extracted into a function ?

File: src/components/Header/Header.styl (hunk 1 above)
Comment (kosssi) [reply src/components/Header/Header.styl]: I remove this part because it's not necessary on this PR 👍

File: src/components/Chart/Tooltip.styl [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    background-color white
+    color dodgerBlue
+    border-radius 2px
+    font-size 0.6875rem
+    padding-left 2px
+    padding-right 2px
+    line-height 1.45
+    white-space nowrap
+
+    // &::after
```
Comment (kosssi) [inline src/components/Chart/Tooltip.styl]: c'est du code pour la petite flèche de la tooltip ça provient d'[ici](https://github.com/cozy/cozy-banks/pull/572/files#diff-6308998e540a440ce24931e2dd4c6db5L4)
